#!/bin/sh /usr/share/dpatch/dpatch-run

## ffmpeg-config patch by Tobias Grimm <tg@e-tobi.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use pkg-config for ffmpeg, fix avcodec.h include and add
## DP: /usr/include/ffmpeg/libavcodec to includes, so that it compiles
## DP: with newer ffmpeg version, wihtout needing to fix the include
## DP: statement

@DPATCH@
diff -urNad vdr-addon-noad-0.7.0~/audiotools.cpp vdr-addon-noad-0.7.0/audiotools.cpp
--- vdr-addon-noad-0.7.0~/audiotools.cpp	2009-12-19 14:08:55.000000000 +0100
+++ vdr-addon-noad-0.7.0/audiotools.cpp	2009-12-30 17:22:49.000000000 +0100
@@ -6,7 +6,7 @@
 #include "mpeg2wrap.h"
 extern "C"
 {
-   #include "avcodec.h"
+   #include <libavcodec/avcodec.h>
 }
 
 #define MIN_LOWVALS 3
diff -urNad vdr-addon-noad-0.7.0~/configure.ac vdr-addon-noad-0.7.0/configure.ac
--- vdr-addon-noad-0.7.0~/configure.ac	2009-12-22 00:21:39.000000000 +0100
+++ vdr-addon-noad-0.7.0/configure.ac	2009-12-30 17:22:49.000000000 +0100
@@ -110,15 +110,8 @@
    withffmpeg="0")
 
 AC_SUBST(HAVE_AVCHEADER,0)
-if test "$withffmpeg" == "1"; then
-  AC_MSG_CHECKING(for libavcodec(ffmpeg))
-  AC_CHECK_FILE(${defprefix}/include/ffmpeg/avcodec.h,
-    [AC_SUBST(libavcodecinc,"-I${defprefix}/include/ffmpeg")],
-    [AC_SUBST(libavcodecinc,"-I/usr/include/ffmpeg")])
-  AC_CHECK_FILE(${defprefix}/lib/libavcodec.so,
-    [AC_SUBST(libavcodecdir,"-L${defprefix}/lib")],
-    [AC_SUBST(libavcodecdir,"-L/usr/local/lib")])
-fi
+AC_SUBST(libavcodecinc,"`pkg-config --cflags libavcodec`")
+AC_SUBST(LIBAVCODEC,`pkg-config --libs libavcodec`)
 
 dnl override include-dir for ffmpeg if user has given a different dir
 AC_SUBST(ffmpegincoverridden,0)
@@ -141,8 +134,11 @@
 dnl ---------------------------------------------------------------------------
 dnl check that libavcodec is usable   
 dnl ---------------------------------------------------------------------------
-AC_SUBST(LIBAVCODEC,"-lavcodec")
-AC_CHECK_LIB([avcodec],[avcodec_init],[AC_SUBST(HAVE_LIBAVCODEC,1) AC_SUBST(LIBS,"$LIBS $LIBAVCODEC ") ])
+LIBS_SAVE="$LIBS"
+LIBS="$LIBS $LIBAVCODEC"
+AC_CHECK_LIB([avcodec],[avcodec_init],[AC_SUBST(HAVE_LIBAVCODEC,1) LIBS_SAVE="$LIBS_SAVE $LIBAVCODEC"])
+LIBS="$LIBS_SAVE"
+AC_SUBST(LIBS,$LIBS)
 
 
 
@@ -231,23 +227,6 @@
 CXXFLAGS='-g -O3'
 
 dnl ---------------------------------------------------------------------------
-dnl check for a usable avcodec.h 
-dnl ---------------------------------------------------------------------------
-if test "$withffmpeg" == "1"; then 
-  dnl check for a usable avcodec.h 
-  dnl have to strip the "-I" from libavcodecinc for doing this
-  AC_SUBST(avcincpath,`echo $libavcodecinc|sed -e s*-I**`)
-  AC_CHECK_HEADER([$avcincpath/avcodec.h],
-    [AC_SUBST(HAVE_AVCHEADER,1)],
-    [AC_SUBST(HAVE_AVCHEADER,0)])
-fi #endif
-if test "$HAVE_LIBAVCODEC" == "1" -a "$HAVE_AVCHEADER" == "1";
-then
-  AC_DEFINE(HAVE_LIBAVCODEC,1,[have libavcodec for audio])
-fi
-
-
-dnl ---------------------------------------------------------------------------
 dnl check for a usable mpeg2.h 
 dnl have to strip the "-I" from mpeginc for doing this
 dnl ---------------------------------------------------------------------------
